tests/chainer_tests/functions_tests/math_tests/test_zeta.py

```diff
@@ -0,0 +1,76 @@
+import unittest
+
+import numpy
+
+import chainer
+import chainer.functions as F
+from chainer import testing
+
+
+@testing.parameterize(*testing.product({
+    'dtype': [numpy.float16, numpy.float32, numpy.float64],
+    'shape': [(), (5, 10)]
+}))
+@testing.inject_backend_tests(
+    None,
+    # CPU tests
+    [
+        {},
+    ]
+    # GPU tests
+    + testing.product({
+        'use_cuda': [True],
+        'cuda_device': [0, 1],
+    })
+    # ChainerX tests
+    + testing.product({
+        'use_chainerx': [True],
+        'chainerx_device': ['native:0', 'cuda:0', 'cuda:1']
+    })
+)
+@testing.with_requires('scipy')
+class TestZeta(testing.FunctionTestCase):
+
+    def setUp(self):
+        self.check_forward_options = {'atol': 1e-3, 'rtol': 1e-3}
+        self.check_backward_options = {'atol': 1e-3, 'rtol': 1e-3}
+        self.check_double_backward_options = {'atol': 1e-3, 'rtol': 1e-3}
+
+        self._x = numpy.random.uniform(self.shape).astype(self.dtype)
+
+    def generate_inputs(self):
+        q = numpy.random.uniform(self.shape).astype(self.dtype)
```

Comment:
Don't you have to restrict the domain of the inputs (rather than sampling from the default U(0, 1)).?

Reply:
Well it is restricted by using `numpy.random.uniform` as q > 0 and x > 1. `shape` defines the range, here its between [5, 10], so it ensures the given restrictions. I'll probably change it to [1, 20] for a broader check. Anything greater than 20 just pushes the result to less than `1e-10`. What do you think?

Reply:
I was not familiar with specifying the lower and upper bounds of the sampling interval each of them using tuples, but that's not how `self.shape` should be used. `self.shape` corresponds to the `size` argument https://docs.scipy.org/doc/numpy-1.15.0/reference/generated/numpy.random.uniform.html no? I also think it's easier to comprehend and consistent with other tests passing `low` and `high` for x and q within their domains using integers/floats?

Reply:
So we'll remove `shape` from parametrization and just pass `low` and `high` in `numpy.random.uniform`?

Reply:
You can leave the parameterization and pass all of `low`, `high` and `shape`.

Reply:
And you can leave the shapes as they were before. The shape shouldn't include negative lengths.

Reply:
I have dropped shape so I'm using `low` and `high`. The negative lows and highs are to check for exceptions

Reply:
Okay you want me to drop negative `low` and `high` and have all parameters in function rather than parameterizing it?

Reply:
Okay here's what I have done. I have retained `shape` for parametrization for generating `numpy.array` and giving range `[low, high)` in the function itself (not parametrized). Is that what was expected?